pepper-apis/src/main/java/org/broadinstitute/ddp/Housekeeping.java

```diff
@@ -511,6 +512,32 @@ private static void respondToGAEStartHook(String envPort) {
             response.status(200);
             return "";
         });
+        Spark.post("/heapdump", (request, response) -> {
+            String postedPassword = request.queryParams("password");
+            if (StringUtils.isNotEmpty(postedPassword)) {
+                String configuredPwd = ConfigManager.getInstance().getConfig().getString(ConfigFile.HEALTHCHECK_PASSWORD);
+                if (!configuredPwd.equals(postedPassword)) {
```

Comment:
Wanted a little bit of security if we put this in production. Healthcheck password was already in config so why not?